### ISSUE TITLE
Add a timeout flag

### ISSUE THREAD

**Issue by nikhiljindal** [GoogleCloudPlatform/k8s-multicluster-ingress#52]
The tool keeps trying indefinitely. We should add a timeout flag so that it gives up after some time.

**Comment by G-Harmon** [comment]
Is there some particular step/steps that you're thinking of? Waiting for some annotation to show up?

**Comment by nikhiljindal** [comment]
good point. the timeout can be per operation timeout or a global timeout. with global timeout, all operations should complete within that time.
